### ISSUE TITLE
Test-Path returns True after another user's process accesses a share

### ISSUE THREAD

**Issue by tjoanp** [PowerShell/PowerShell#6474]
Steps to reproduce
------------------
Logged in as user1, who doesn't have access to restrictedfolder, running pwsh:

> Test-Path \\\server\share\restrictedfolder
Returns False    <--- This is correct

Open cmd.exe as user2, who have access to restrictedfolder, and run:

> dir \\\server\share

Switch back to the pwsh window, still running as user1

> Test-Path \\\server\share\restrictedfolder
Returns True   <--- This is incorrect

Still in pwsh, run 
> dir \\\server\share
> Test-Path \\\server\share\restrictedfolder
Returns False    <--- This is now correct again

Expected behavior
-----------------
Test-Path Should return False the whole time

Actual behavior
---------------
Test-Path returns True for user1 after user2, who has more permissions in the file system. has accessed the share

This issue occurs even when running Test-Path against subfolders, for example \\server\share\folder\restrictedfolder. I can however still "fix" it by running dir \\\server\share.

Environment data
----------------
```
Name                           Value
----                           -----
PSVersion                      6.0.2
PSEdition                      Core
GitCommitId                    v6.0.2
OS                             Microsoft Windows 6.3.9600
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

```

**Comment by BrucePay** [comment]
@tjoanp What happens on your system if you use cmd.exe for both users i.e. run cmd.exe and do a dir on the restricted folder instead of running PowerShell and doing a Test-Path?

**Comment by tjoanp** [comment]
That works as expected, i.e. user1 never sees more folders than the permissions allows.

**Comment by tjoanp** [comment]
Perhaps I'm looking at a .Net issue?  This also returns True:
> [System.IO.Directory]::Exists("\\server\share\restrictedfolder")

**Comment by iSazonov** [comment]
@tjoanp Please open Issue in CoreFX repo - we could get the fix in 2.1.0.
